modules/internal/ChapelDistribution.chpl

```diff
@@ -394,5 +582,63 @@ module ChapelDistribution {
     proc dsiSupportsBulkTransferInterface() param return false;
     proc doiCanBulkTransferStride() param return false;
   }
-  
+
+  class BaseSparseArr: BaseArr {
+    type eltType;
+    param rank : int;
+    type idxType;
+
+    var dom; /* : DefaultSparseDom(?); */
+    /*var dataDom = {1..0};*/
+    var data: [dom.nnzDom] eltType;
```

Comment:
I'd leave it here because of series of intertwined issues.

If we moved it to `BaseSparseArrImpl` then, it would be using a generic field of the parent class. Which means we'd need something roughly like this:

```
class BaseSparseArrImpl: BaseSparseArr {
   // ....
   var dataDom: {1..0};
   var data: [dataDom] eltType;
   // ....
}

class DefaultSparseArr: BaseSparseArrImpl {
   // ....
   proc DefaultSparseArr(.... , dom, ....) {
      this.dataDom = dom.nnzDom; //this is a domain assignment, so not what we want
   }
   // ....
}
```

Another way to make it work is to move `dom` to `BaseSparseArrImpl`, but that would be against the idea of `BaseSparseArrImpl`. Moreover, as this gets refactored further `dom` should move even higher in the hierarchy, I believe.